simulation/traffic_simulator/include/traffic_simulator/math/hermite_curve.hpp

```diff
@@ -56,6 +56,7 @@ class HermiteCurve
   double getMaximum2DCurvature() const;
   double getLength(size_t num_points) const;
   double getLength() const { return length_; }
+  void setLength(double length) { length_ = length; }
```

Comment:
Length should be calculated in hermite curve class.
Why do you add setLength function?

Reply:
This is needed for curve trimming. This PR aims to avoid spline recalculation because this operation is very costly. Instead, the `trajectory` is calculated as a section of `reference_trajectory`. To calculate the subspline (which is just a part of the spline) you need to determine which curves should be included in the subspline. The middle curves are taken as a whole, but the first one and the last one need to be trimmed (because the start point of the subspline can be in the middle of the curve and we need only a part of it to construct the subspline). The curves should not change their shapes, but only change their length. So `setLength` is used to shorten the curves without the need to recalculate them from scratch (because it is costly as well).

Reply:
I see, but I think this function is dangerous because of this is a public function.
And also, this function does not changes starting point and ending point of hermite curve it breaks the hermite curve class.

Reply:
Yes, I agree this solution is not perfect. I will try to come up with a better idea. If you have any suggestions on how to implement hermite curve trimming I would really appreciate it.

Reply:
I think problem is setting length.
Can you use constructor of HermiteCurve class?

Reply:
We discussed this problem internally. We agree that length setting is a problem. We came up with a solution: we will introduce the `CatmullRomSubspline` class. It stores a reference to the original `CatmullRomSpline`, `start_s` and `end_s`. It provides an interface for accessing the underlying spline as it was trimmed (as it would start in `start_s` and end in `end_s`). The underlying spline would not be changed in any way. Therefore, the dangerous functions `HermiteCurve::setLength` and `CatmullRomSpline::getSubspline` will no longer be needed and will be removed.

What do you think about this approach?

Reply:
Do you mean you trimmed spline in CatmullRomSubspline class not in HermiteCurve class?
And s value of the Hermite curve class is always from 0 to it's length?
If so, I think it is a good idea!

Reply:
I believe it's even better - I do not trim the spline at all. I use the `CatmullRomSubspline` class to access the spline as it was trimmed, but the underlying spline is not modified at all.